### PR TITLE
Implement View Photos carousel WIP

### PR DESCRIPTION
**Pull request by woeltjen**
## Description
Continue rebrand of the listing page; implement View Photos

## How to Test
1. Navigate to a listing
2. Click View Photos
3. Expect to see carousel showing photos at higher resolution

Which devices did you test on?
- [x] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [ ] Safari iPhone
- [ ] Chrome Android

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
1. Contact Host functionality
2. Booking Card layout
3. Booking Card functionality

### CODE REVIEW

**Comment by woeltjen** [comment]
image

Trying to use `Carousel` from `reactstrap` for this; the existing `ListingCarousel` uses a lot of custom CSS around `react-slick`. Main issues right now are legibility of arrows and caption text. Having some difficulty customizing these. So, opening this PR as WIP - does this seem like a direction worth going in, or should we just carry over our existing Carousel?

**Comment by kcvan** [comment]
i think our existing carousel also has some issues with the styling and arrows hiding behind the image.  but i feel your pain, reactstrap's carousel component will need some tweaking in order to get things looking nice.  I would try and tweak reactstrap if you can, you might need to add some custom classes in the customStyles file but i think it might be easier to do that than dive back into react-slick

**Comment by woeltjen** [comment]
Removed WIP status. I'm not totally happy with the visibility of the left/right arrows, but it's only an issue on brighter images, and presumably the theme authors knew what they were doing...

File: src/pages/listing/ListingCarousel.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { Carousel, CarouselCaption, CarouselControl, CarouselItem } from 'reactstrap';
+
+import LazyImage from 'components/shared/LazyImage';
+
+interface Props {
+  photos: string[];
+}
+
+// TODO: add a ref so that keyboard can be used immediately when carousel opens
+// see: https://github.com/akiran/react-slick/issues/738
```
Comment (jerminatorhits) [inline src/pages/listing/ListingCarousel.tsx]: comment can be removed

File: src/pages/listing/ListingGallery.tsx [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
-  <Row className="w-100 height-60vh px-0 mx-0 position-relative">
+const ListingGallery = ({ listingPicUrl, photos }: Listing) => {
+  const [isOpen, setOpen] = React.useState<boolean>(false);
+  return <Row className="w-100 height-60vh px-0 mx-0 position-relative">
     <ImageGrid images={[listingPicUrl, ...photos]} />
-    <Button className="position-absolute bottom-0 right-0 m-4">
+    <Button className="position-absolute bottom-0 right-0 m-4" onClick={() => setOpen(true)}>
       View Photos <span className="fas fa-camera"></span>
     </Button>
+    <Modal centered size="lg" isOpen={isOpen}>
```
Comment (jerminatorhits) [inline src/pages/listing/ListingGallery.tsx]: could also add a `toggle={() => setOpen(false)}` so users can click out of modal

File: src/pages/listing/ListingCarousel.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+const ListingCarousel = ({ photos }: Props) =>  {
+  const [index, setIndex] = React.useState<number>(0);
+  const next = () => setIndex((index + 1) % photos.length);
+  const previous = () => setIndex((index - 1 + photos.length) % photos.length);
+  return <Carousel activeIndex={index} next={next} previous={previous}>
```
Comment (jerminatorhits) [inline src/pages/listing/ListingCarousel.tsx]: `<Carousel>` has a prop `interval={false}` that turns off the autoscrolling allowing users to click through photos at their own pace. tried `autoPlay={false}` but it didn't work =/

**Comment by kcvan** [comment]
i think it looks good overall.  the only thing i can think of is to add a custom class to make the carousel full width on mobile, and change the `left: 0` and `right: 0` of the carousel buttons to different positions based on the screen type.

File: src/pages/listing/ListingCarousel.tsx (hunk 1 above)
Comment (woeltjen) [reply src/pages/listing/ListingCarousel.tsx]: 🙏

File: src/pages/listing/ListingGallery.tsx (hunk 2 above)
Comment (woeltjen) [reply src/pages/listing/ListingGallery.tsx]: Oooh fancy!

**Comment by kcvan** [comment]
getting some strange heights on my end:
x large:
screen shot 2019-03-05 at 1 06 37 pm

large:
screen shot 2019-03-05 at 1 06 50 pm

tablet:
screen shot 2019-03-05 at 1 07 06 pm

mobile:
screen shot 2019-03-05 at 1 06 59 pm

**Comment by woeltjen** [comment]
Sorry, should have WIP'd this.

Actually, just going to close this. This has become an unreasonable time sink. Will look for a drop-in solution.

**Pull request closed without merging**